lib/stages/pulsarPostProcess.cpp

```diff
@@ -85,19 +85,28 @@ void pulsarPostProcess::fill_headers(unsigned char* out_buf, struct VDIFHeader*
           time_now->tv_nsec);
     for (uint i = 0; i < _num_packet_per_stream; ++i) { // 16 or 80 frames in a stream
         uint64_t fpga_now = (fpga_seq_num + _timesamples_per_pulsar_packet * i);
-        vdif_header->seconds = time_now->tv_sec - unix_offset;
-        vdif_header->data_frame =
+        psr_header->seconds = time_now->tv_sec - unix_offset;
+        psr_header->data_frame =
             (time_now->tv_nsec / 1.e9) / (_timesamples_per_pulsar_packet * fpga_s);
 
         for (uint f = 0; f < freqloop; ++f) {
-            vdif_header->thread_id = thread_ids[f];
+            // Before looping over pulsar beams, load freq labels into header.
+            // As of 7 September 2020, load one of four labels into thread_id, 
+            // and pack the other three into the 32-b EUD3 attribute.
+            // TODO: redefine pulsar packet structure to not use VDIF def, instead use 
+            // structures with dimensions defined by telescope-config variables.
+            if (f == 0) {
+                psr_header->thread_id = thread_ids[f];
+            else {
+                psr_header->eud3 += (thread_ids[f] << (f - 1) * 10);
```

Comment:
The `+=` here requires that `eud3` contains zero at the start of this loop.  However that's not a guarantee, frames are not zeroed between uses.  You can likely just zero this in the `f == 0` case, or add another case to this if for `f == 1`.

Reply:
ah, great catch andre; added a zeroing statement within the if-statement.